nbgitpuller/plugins/README.md

```diff
@@ -0,0 +1,23 @@
+# [nbgitpuller download plugins](https://github.com/jupyterhub/nbgitpuller)
+
+`nbgitpuller` download plugins enable users to download compressed
+archives(zip or tar-compatible) into jupyter hubs from any publicly accessible URL
+including from services such as Google Drive and Dropbox. Each plugin in this directory
+includes a README file describing the format of the URL expected from each provider.
+
+You can install some or all of the plugins into your environment. They are automatically
+discovered by the system; we used pluggy(https://pluggy.readthedocs.io/en/stable/) to handle
+the loading and implementation of these plugins.
```

Comment:
These are instructions related to plugins, no matter if they are downloader plugins or something else and could be extracted to a higher level.

Reply:
> @sean-morris Thanks for your answers! I think the decision on whether integration tests are required or not is one factor in whether the plugins should be moved to a different repo. Having them in a separate repo makes it clear to nbgitpuller developers that their responsibility is to maintain the API of nbgitpuller and not worry about the implementations of these plugins. Another benefit of putting the plugins in a separate repo is it's easier to add or update integration tests now or in future since the config and any secrets are seperate.
> 
> A further consideration is whether you want to version/release plugins independently of nbgitpuller- if so it's a lot easier to do this in a separate repo, otherwise the CI publish workflows get complicated.

@manics I will pull the plugins out of nbgitpuller -- makes sense. I am assuming then that each plugin gets its own repo; I would name(!) them:
- nbgitpuller-downloader-generic-web
- nbgitpuller-downloader-googledrive
- nbgitpuller-downloader-dropbox

Reply:
@consideRatio yes I don't know what I was thinking in the first round of documentation here. I was completely documenting my own process of understanding pluggy(!) not what others would need to do to develop a downloader plugin against nbgitpuller.